content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md

```diff
@@ -176,20 +176,24 @@ policy during rolling upgrades.
 : With this feature gate enabled, when joining a new control plane node, a new etcd member will be created
 as a learner and promoted to a voting member only after the etcd data are fully aligned.
 
-`PublicKeysECDSA`
-: Can be used to create a cluster that uses ECDSA certificates instead of the default RSA algorithm.
-Renewal of existing ECDSA certificates is also supported using `kubeadm certs renew`, but you cannot
-switch between the RSA and ECDSA algorithms on the fly or during upgrades. Kubernetes
-{{< skew currentVersion >}} has a bug where keys in generated kubeconfig files are set use RSA
-despite the feature gate being enabled. Kubernetes versions before v1.31 had a bug where keys in generated kubeconfig files
-were set use RSA, even when you had enabled the `PublicKeysECDSA` feature gate.
+`NodeLocalCRISocket`
+: With this feature gate enabled, kubeadm will read/write the CRI socket for each node from/to the file
+`/var/lib/kubelet/instance-config.yaml` instead of reading/writing it from/to the annotation
+`kubeadm.alpha.kubernetes.io/cri-socket` on the Node object. The new file is applied as an instance
+configuration patch, before any other user managed patches are applied when the `--patches` flag
+is used. It contains a single field `containerRuntimeEndpoint` from the
+[KubeletConfiguration file format](/docs/reference/config-api/kubelet-config.v1beta1/).
```

Comment:
For upgrade, we will read from `kubeadm-flags.env` if `instance-config.yam` does not exist.

Should we mention it?

Reply:
adding:
> If the feature gate is enabled during upgrade, but the file `/var/lib/kubelet/instance-config.yaml` does not exists yet,
kubeadm will attempt to read the CRI socket value from the file `/var/lib/kubelet/kubeadm-flags.env`.